client/src/components/overview/SizeSelector.jsx

```diff
@@ -1,50 +1,105 @@
 import React from 'react';
 
-const SizeSelector = (props) => {
-  const items = [];
-  for (let i = 0; i < props.skus.length; i++) {
-    items.push({
-      sku: props.skus[i],
-      size: props.sizes[i]
-    });
+class SizeSelector extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.onChange = this.onChange.bind(this);
+
+    this.state = {
+      currentSize: ''
+    }
+  }
+
+  componentDidUpdate() {
+    const items = this.getItems(this.props.skus, this.props.sizes);
```

Comment:
I think the `Array.map` method will provide an index, so this could be `const items = this.props.skus.map((sku, i) => ({sku: sku, size: this.props.sizes[i]}));`